src/lib/component/partial/HandlersList/HandlersListItemDetails.js

```diff
@@ -0,0 +1,93 @@
+import React from "/vendor/react";
+import PropTypes from "prop-types";
+import gql from "/vendor/graphql-tag";
+
+import { Code, CodeHighlight } from "/lib/component/base";
+import { Maybe, NamespaceLink } from "/lib/component/util";
+
+const HandlersListItemDetails = ({ handler }) => (
+  <React.Fragment>
+    <div>
+      {`Type: `}
+      <strong>{handler.type}</strong>
+    </div>
+
+    <Maybe value={handler.type === "pipe"}>
+      {() => (
+        <React.Fragment>
+          {`Command: `}
+          <CodeHighlight
+            language="bash"
+            code={handler.command}
+            component={Code}
+          />
+        </React.Fragment>
+      )}
+    </Maybe>
+
+    <Maybe value={handler.type === "set"}>
+      {() => (
+        <React.Fragment>
+          {`Handlers: [ `}
+          {handler.handlers.map((hd, idx, arr) => (
+            <React.Fragment key={hd.name}>
+              <NamespaceLink
+                namespace={hd.namespace}
+                to={`/handers/${hd.name}`}
+              >
+                <CodeHighlight
+                  language="json"
+                  code={`${hd.name}`}
+                  component={Code}
+                />
+              </NamespaceLink>
+              {idx < arr.length - 1 ? ", " : ""}
+            </React.Fragment>
+          ))}
+          {` ]`}
+        </React.Fragment>
+      )}
+    </Maybe>
+
+    <Maybe value={handler.type === "tcp" || handler.type === "udp"}>
+      {() => (
+        <React.Fragment>
+          {`Port: `}
+          <CodeHighlight
+            language="json"
+            code={`${handler.socket.port}`}
+            component={Code}
+          />
+          {` Host: `}
+          <CodeHighlight
+            language="json"
+            code={handler.socket.host}
+            component={Code}
+          />
```

Comment:
I wonder if we could instead do `{scheme}://{host}:{port}`. eg. `tcp://123.2.3.1:3838`, `udp://8.3.1.35:3131`, etc.

Reply:
OH yes! that's much nicer. Will do will do -- thanks!